File: VB/WinDesignerDashboardState/DesignerForm1.vb
Imports Microsoft.VisualBasic
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardWin
Imports System
Imports System.Collections.Generic
Imports System.Xml.Linq

Namespace WinDesignerDashboardState
	Partial Public Class DesignerForm1
		Inherits DevExpress.XtraBars.Ribbon.RibbonForm
		Private Const path As String = "..\..\Dashboards\dashboardWithSavedState.xml"
		Public Sub New()
			InitializeComponent()
			dashboardDesigner.CreateRibbon()
			dashboardDesigner.UseNeutralFilterMode = True
			dashboardDesigner.LoadDashboard(path)
		End Sub

		Public Function CreateDashboardState() As DashboardState
			Dim state As New DashboardState()
			'Clears selections in the LisxBox item.
			state.Items.Add(New DashboardItemState("listBoxDashboardItem1") With {.MasterFilterValues = New List(Of Object())()})
			'The lines below clears all values if neutral filter mode is disabled:
			'state.Items.Add(new DashboardItemState("listBoxDashboardItem1") {
			'    MasterFilterValues = null
			'});
			Return state
		End Function
		Private Sub dashboardDesigner_CustomizeDashboardTitle(ByVal sender As Object, ByVal e As DevExpress.DashboardWin.CustomizeDashboardTitleEventArgs) Handles dashboardDesigner.CustomizeDashboardTitle
			Dim resetStateItem As New DashboardToolbarItem("Reset State", New Action(Of DashboardToolbarItemClickEventArgs)(Function(args) AnonymousMethod1(args)))
			resetStateItem.Caption = "Reset Dashboard State"
			e.Items.Add(resetStateItem)
		End Sub
		
		Private Function AnonymousMethod1(ByVal args As Object) As Boolean
			dashboardDesigner.SetDashboardState(CreateDashboardState())
			Return True
		End Function
	End Class
End Namespace

File: CS/WinDesignerDashboardState/DesignerForm1.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardWin;
using System;
using System.Collections.Generic;
using System.Xml.Linq;

namespace WinDesignerDashboardState
{
    public partial class DesignerForm1: DevExpress.XtraBars.Ribbon.RibbonForm {
        const string path = @"..\..\Dashboards\dashboardWithSavedState.xml";
        public DesignerForm1() {
            InitializeComponent();
            dashboardDesigner.CreateRibbon();
            dashboardDesigner.UseNeutralFilterMode = true;
            dashboardDesigner.LoadDashboard(path);
        }

        public DashboardState CreateDashboardState() {
            DashboardState state = new DashboardState();
            //Clears selections in the LisxBox item.
            state.Items.Add(new DashboardItemState("listBoxDashboardItem1") {
                MasterFilterValues = new List<object[]>()
            });
            //The lines below clears all values if neutral filter mode is disabled:
            //state.Items.Add(new DashboardItemState("listBoxDashboardItem1") {
            //    MasterFilterValues = null
            //});
            return state;
        }
        private void dashboardDesigner_CustomizeDashboardTitle(object sender, DevExpress.DashboardWin.CustomizeDashboardTitleEventArgs e) {
            DashboardToolbarItem resetStateItem = new DashboardToolbarItem("Reset State",
              new Action<DashboardToolbarItemClickEventArgs>((args) => {
                  dashboardDesigner.SetDashboardState(CreateDashboardState());
              }));
            resetStateItem.Caption = "Reset Dashboard State";
            e.Items.Add(resetStateItem);
        }
    }
}
